addons/pkg-npm/common/src/main/java/org/commonjava/indy/pkg/npm/content/group/PackageMetadataMerger.java

```diff
@@ -0,0 +1,150 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.commonjava.indy.pkg.npm.content.group;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.apache.commons.io.IOUtils;
+import org.commonjava.indy.IndyWorkflowException;
+import org.commonjava.indy.model.core.Group;
+import org.commonjava.indy.model.core.StoreKey;
+import org.commonjava.indy.model.core.io.IndyObjectMapper;
+import org.commonjava.indy.pkg.npm.model.PackageMetadata;
+import org.commonjava.maven.galley.model.Transfer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.inject.Instance;
+import javax.inject.Inject;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import static org.apache.commons.io.IOUtils.closeQuietly;
+import static org.commonjava.indy.util.LocationUtils.getKey;
+
+public class PackageMetadataMerger
+{
+    public static final String METADATA_NAME = "package.json";
+
+    @Inject
+    private Instance<PackageMetadataProvider> metadataProviderInstances;
+
+    private List<PackageMetadataProvider> metadataProviders;
+
+    protected PackageMetadataMerger()
+    {
+    }
+
+    public PackageMetadataMerger( Iterable<PackageMetadataProvider> providers )
+    {
+        metadataProviders = new ArrayList<>();
+        providers.forEach( provider -> metadataProviders.add( provider ) );
+    }
+
+    @PostConstruct
+    public void setupCDI()
+    {
+        metadataProviders = new ArrayList<>();
+        if ( metadataProviderInstances != null )
+        {
+            metadataProviderInstances.forEach( provider -> metadataProviders.add( provider ) );
+        }
+    }
+
+    public byte[] merge( final Collection<Transfer> sources, final Group group, final String path )
+    {
+        Logger logger = LoggerFactory.getLogger( getClass() );
+        logger.debug( "Generating merged metadata in: {}:{}", group.getKey(), path );
+
+        InputStream stream = null;
+
+        boolean merged = false;
+
+        final PackageMetadata packageMetadata = new PackageMetadata();
+        final IndyObjectMapper mapper = new IndyObjectMapper( true );
+
+        for ( final Transfer src : sources )
+        {
+            if ( !src.exists() )
+            {
+                continue;
+            }
+
+            try
+            {
+                stream = src.openInputStream();
+                String content = IOUtils.toString( stream );
+                logger.debug( "Adding in metadata content from: {}\n\n{}\n\n", src, content );
+
+                PackageMetadata md = mapper.readValue( content, PackageMetadata.class );
+
+                packageMetadata.merge( md );
+                merged = true;
```

Comment:
merged should be:

```
    merged = packageMetadata.merge( md ) || merged;
```

This way, you maintain the merged flag as true if it was already set, but you only set it to true if the packageMetadata instance is changed by a merge operation.